pylib/mozhginfo/mozhginfo/mozhginfo.py

```diff
@@ -0,0 +1,167 @@
+    the following best practices:
+
+    * Query by push ID, not by changeset or date.
+    * Always specify a startID and endID.
+    * Try to avoid full if possible.
+    * Always use the latest format version.
+    * Don't be afraid to ask for a new pushlog feature to make your life easier.
+"""
+import logging
+
+import requests
+
+
+LOG = logging.getLogger('pushlog_client')
+JSON_PUSHES = "%(repo_url)s/json-pushes"
+VALID_CACHE = {}
+
+
+class PushlogAuthError(Exception):
+    pass
+
+
+def query_revisions_range(repo_url, from_revision, to_revision, version=2, tipsonly=1):
+    """
+    Return an ordered list of revisions (by date - oldest (starting) first).
+
+    repo           - represents the URL to clone a repo
+    from_revision - from which revision to start with (oldest)
+    to_revision   - from which revision to end with (newest)
+    version        - version of json-pushes to use (see docs)
+    """
+    revisions = []
+    url = "%s?fromchange=%s&tochange=%s&version=%d&tipsonly=%d" % (
+        JSON_PUSHES % {"repo_url": repo_url},
+        from_revision,
+        to_revision,
+        version,
+        tipsonly
+    )
+    LOG.debug("About to fetch %s" % url)
+    req = requests.get(url)
+    pushes = req.json()["pushes"]
+    # json-pushes does not include the starting revision
+    revisions.append(from_revision)
+    for push_id in sorted(pushes.keys()):
+        # Querying by push ID is perferred because date ordering is
+        # not guaranteed (due to system clock skew)
+        # We can interact with self-serve with the full char representation
+        revisions.append(pushes[push_id]["changesets"][-1])
+
+    return revisions
+
+
+def query_pushid_range(repo_url, start_id, end_id, version=2):
+    """
+    Return an ordered list of revisions (newest push id first).
+
+    repo     - represents the URL to clone a repo
+    start_id - from which pushid to start with (oldest)
+    end_id   - from which pushid to end with (most recent)
+    version  - version of json-pushes to use (see docs)
+    """
+    revisions = []
+    url = "%s?startID=%s&endID=%s&version=%s&tipsonly=1" % (
+        JSON_PUSHES % {"repo_url": repo_url},
+        start_id - 1,  # off by one to compensate for pushlog as it skips start_id
+        end_id,
+        version
+    )
+    LOG.debug("About to fetch %s" % url)
+    req = requests.get(url)
+    pushes = req.json()["pushes"]
+    # pushes.keys() is a list of strings which we need to map to integers
+    # We use reverse in order to return list sorted from newest to oldest push id
+    for push_id in sorted(map(int, pushes.keys()), reverse=True):
+        # Querying by push ID is preferred because date ordering is
+        # not guaranteed (due to system clock skew)
+        # We can interact with self-serve with the 12 char representation
+        revisions.append(pushes[str(push_id)]["changesets"][0])
+
+    return revisions
+
+
```

Comment:
Where is query_revisions_range_from_revision_before_and_after? We use it in trigger.py